### ISSUE TITLE
Generated code fails the `clippy::str_to_string` lint

### ISSUE THREAD

**Issue by dnbln** [graphql-rust/juniper#1146]
**Describe the bug**
The generated code fails the `clippy::str_to_string` lint.

**To Reproduce**

<details open><summary><code>src/main.rs</code></summary>

```rs
#![deny(clippy::str_to_string)]

struct A;

#[juniper::graphql_object]
impl A {
    fn f() -> &'static str {
        "a"
    }
}


fn main() {
}
```

</details>

<details><summary><code>Cargo.toml</code></summary>

```toml
# Cargo.toml
[package]
name = "x"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
juniper = "0.15.10"
```
</details>

<details open><summary><code>cargo clippy</code> output</summary>

```
Checking x v0.1.0 (C:\Users\Dinu\proj\x)
error: `to_string()` called on a `&str`
 --> src\main.rs:5:1
  |
5 | #[juniper::graphql_object]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: consider using `.to_owned()`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#str_to_string
note: the lint level is defined here
 --> src\main.rs:1:9
  |
1 | #![deny(clippy::str_to_string)]
  |         ^^^^^^^^^^^^^^^^^^^^^
  = note: this error originates in the attribute macro `juniper::graphql_object` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `x` due to previous error
```

</details>

**Expected behavior**
The generated code to pass the lint, or at least ignore it.

**Additional context**

<details><summary>
<code>juniper</code> and <code>juniper_codegen</code> in <code>Cargo.lock</code>
</summary>

```toml
# Cargo.lock
[[package]]
name = "juniper"
version = "0.15.11"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "52adf17d43d0b526eed31fac15d9312941c5c2558ffbfb105811690b96d6e2f1"
dependencies = [
 "async-trait",
 "bson",
 "chrono",
 "fnv",
 "futures",
 "futures-enum",
 "graphql-parser",
 "indexmap",
 "juniper_codegen",
 "serde",
 "smartstring",
 "static_assertions",
 "url",
 "uuid",
]

[[package]]
name = "juniper_codegen"
version = "0.15.9"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "aee97671061ad50301ba077d054d295e01d31a1868fbd07902db651f987e71db"
dependencies = [
 "proc-macro-error",
 "proc-macro2",
 "quote",
 "syn",
]
```
</details>

**Comment by ShadowWalker98** [comment]
Can I try this issue? I am new to Rust though.

**Comment by LegNeato** [comment]
Of course!

**Comment by tyranron** [comment]
This is already fixed on `master` and will be released in 0.16.
